### ISSUE TITLE
print_summary() miss calculates posterior summary statistics for adstock and hill_adstock model

### ISSUE THREAD

**Issue by ohad-monday** [google/lightweight_mmm#96]
Hi
bumped into it when running a few models, and noticed this weird thing.
Ran it over the demo data just to make sure.
so, if you run model with adstock or hill__adstock:
mmm = lightweight_mmm.LightweightMMM(model_name="adstock") 

print_summary() will bring weird values for coefficients (percentiles 5/95 etc) - the numbers don't match the numbers in the plot
plot.plot_bars_media_metrics

                            mean       std    median      5.0%     95.0%     n_eff     r_hat
coef_extra_features[0]      0.02      0.01      0.02      0.00      0.04    908.31      1.00
         coef_media[0]      0.12      0.13      0.07      0.00      0.32    117.19      1.02
         coef_media[1]      0.16      0.05      0.16      0.09      0.24    574.67      1.01
         coef_media[2]      0.09      0.06      0.09      0.00      0.16    420.97      1.01
         coef_trend[0]     -0.01      0.01     -0.01     -0.02      0.00     86.79      1.04
            expo_trend      0.59      0.10      0.55      0.50      0.71    470.10      1.01
           exponent[0]      0.91      0.08      0.94      0.80      1.00    936.84      1.00
           exponent[1]      0.91      0.08      0.93      0.80      1.00   1569.54      1.00
           exponent[2]      0.90      0.09      0.93      0.78      1.00    982.93      1.00
gamma_seasonality[0,0]      0.02      0.95      0.02     -1.46      1.59   1092.37      1.00
gamma_seasonality[0,1]     -0.10      0.55     -0.03     -0.92      0.75    378.44      1.01
gamma_seasonality[1,0]     -0.02      0.00     -0.02     -0.02     -0.01   1528.42      1.00
gamma_seasonality[1,1]     -0.00      0.00     -0.00     -0.01      0.01   1303.82      1.00
          intercept[0]      0.85      0.55      0.78      0.00      1.63    375.15      1.01
         lag_weight[0]      0.86      0.22      0.99      0.48      1.00    138.94      1.03
         lag_weight[1]      0.33      0.11      0.32      0.16      0.52    349.45      1.01
         lag_weight[2]      0.40      0.23      0.35      0.03      0.75     39.06      1.06

<img width="768" alt="image" src="https://user-images.githubusercontent.com/61185809/199091179-f546716d-174c-4dd1-83ff-608df43fdfb4.png">

when calculating the metrics manually, i'm getting the same numbers as in the plot.plot_bars_media_metrics

`def calc_posteriors_statistics(mmm_model, target_scaler, cost_scaler, lower_ci=5, upper_ci=95):

    posteriors = mmm_model.get_posterior_metrics(target_scaler=target_scaler, cost_scaler=cost_scaler)[0]
    statistics_dict = {
                       'perc_5': str(jnp.percentile(posteriors, lower_ci, axis=0).tolist()),
                       'perc_95': str(jnp.percentile(posteriors, upper_ci, axis=0).tolist()),
                       'mean': str(jnp.mean(posteriors, axis=0).tolist()),
                       'median': str(jnp.median(posteriors, axis=0).tolist())}

    return statistics_dict

calc_posteriors_statistics(mmm, target_scaler, cost_scaler)`

<img width="573" alt="image" src="https://user-images.githubusercontent.com/61185809/199091248-a27da0c3-15d9-4ec2-9578-2269d16e24cd.png">

**Comment by michevan** [comment]
Hi @ohad-monday, sorry, but can you clarify where the numbers aren't matching? 

If I understand correctly, I don't think any of the numbers in LightweightMMM.print_summary() are supposed to match the numbers in LightweightMMM.get_posterior_metrics(). The fractional media contributions for each channel are functions of beta and the shape parameters, but you have to marginalize over the posterior distributions for all the parameters in the model in order to get the fractional media contributions. Does that make sense?

**Comment by ohad-monday** [comment]
Hi @michevan 
hope to be clear now:)

Shortly, if you run lightweight_mmm.LightweightMMM() with adstock/hill_adtock on the demo data, and than run lightweight_mmm.print_summary() the numbers you gonna get from the print_summary() won't match the numbers in the plot you get from plot.plot_bars_media_metrics
The numbers should match (coefficients percentiles 5% and 95% are calculated in print_summary and are shown in the plot) - well they don't match.


I created a customized function that calculates these metrics from the posteriors, and the numbers i get match the numbers in the plot, vs the print_summary() numbers that don't match.


I would say the the numbers for carryover print_summary() model do match the plot. adstock and hill_adtock from the otherhand, don't.

this is the function i used to calculate percentiles 5% and 95% for channels estimated media contribution:
def calc_posteriors_statistics(mmm_model, target_scaler, cost_scaler, lower_ci=5, upper_ci=95):

              posteriors = mmm_model.get_posterior_metrics(target_scaler=target_scaler, cost_scaler=cost_scaler)[0]
              statistics_dict = {
                   'perc_5': str(jnp.percentile(posteriors, lower_ci, axis=0).tolist()),
                   'perc_95': str(jnp.percentile(posteriors, upper_ci, axis=0).tolist()),
                   'mean': str(jnp.mean(posteriors, axis=0).tolist()),
                   'median': str(jnp.median(posteriors, axis=0).tolist())}
return statistics_dict


you can see in the plot for example that for media channel_0 the 95% should be around ~16%, while in the print_summary() output it calculated ~32%

                        mean  std    median      5.0%     95.0%     n_eff     r_hat
coef_media[0] 0.12 0.13 0.07 0.00 0.32 117.19 1.02
coef_media[1] 0.16 0.05 0.16 0.09 0.24 574.67 1.01
coef_media[2] 0.09 0.06 0.09 0.00 **0.16** 420.97 1.01


The above is not relevant for carryover model

**Comment by pabloduque0** [comment]
Hello @ohad-monday !

@michevan is right in the sense that those numbers should not necessarily match.

`get_posterior_metrics` calculates media contribution/effect which is not exactly the value of the coefficient since we have media transformation which play a factor in the former.

Hope it helps!

**Comment by ohad-monday** [comment]
@pabloduque0 
So just for carryover it was the same by coincidence? 

my confusion then:)

thanks

**Comment by pabloduque0** [comment]
Carryover model might in your case be learning less severe transformations which result on the numbers matching.

Hope it helps!
